radixdlt/babylon-wallet-android
# [ABW-3200] Sanitize decimal input and use locale based formatting for decimal inputs

## Description
- changes step for increasing/decreasing guarantees on transfer screen
- used new methods from sargon to parse decimal192 from text input/convert decimal192 to text using system locale
- sanitize text field input to only allow digits and decimal separator specific for current locale. Unfortunately on Android, even if current locale separator is `,` keyboard also allows to input `.`, and other way around

## How to test

1. Before the change try to increase/decrease guarantees value on transaction review screen. Step is 0.1 percent.
2. Before the change try to increase/decrease guarantee amount in different locales, for example UK, Greek, Polish. Notice that amount is reset to 0 due to incorrect parsing.
3. Try to copy & paste decimals that use use separator from different locale, for example in poland, where decimal separator is `,` when I copy 10.58, it get pasted as 1058. 

## PR submission checklist
- [ ] I have decimal input in transaction guarantees/fees/amount input
